CharlieBackend.Core/Entities/StudentGroup.cs

```diff
@@ -6,7 +6,7 @@ namespace CharlieBackend.Core.Entities
     public partial class StudentGroup : BaseEntity
     {
 
-        public long CourseId { get; set; }
+        public long? CourseId { get; set; }
```

Comment:
I think here you need to use macroses #nullable enable and #nullable disable after that.
Compiler should tell you about that.